bgzf.c

```diff
@@ -2205,8 +2205,15 @@ static inline int64_t bgzf_seek_common(BGZF* fp,
                 abort();  // Should not get to any other state
             }
         } while (fp->mt->command != SEEK_DONE);
+
         fp->mt->command = NONE;
 
+        if (fp->mt->errcode) {
+            fp->errcode |= BGZF_ERR_IO;
```

Comment:
Also `errno = fp->mt->errcode` to propagate `errno` over to the calling thread (if this multithreading works the way I assume it does!).

Reply:
Your probably right.